### DISCUSSION TITLE
Please add existing Tags to full text search results

### DISCUSSION THREAD

**Discussion by xFFF6** [karakeep-app/karakeep#198] [Ideas]
When I'm using the full text search, I'd love to see the existing matching tags in search results instead of switching to the Tags site. This would make the search more effective.

**Comment by kamtschatka** [comment]
I don't understand what you mean. When doing a full text search, you already see the tags on the bookmarks? Do you mean you want to have search functionality to tags?
Or a separate section where you see all the tags of all the bookmarks?

**Reply by xFFF6** [reply]
I have use cases where I want to get search results, when searching for specific words, what are not part of the bookmarked page without need to go to the Tag section of hoarder.
For example:
kicad is part of the bookmarked page and will be found:
![image](https://github.com/hoarder-app/hoarder/assets/80603288/4749e8f4-cb1d-4903-b2a3-3bc18ab58353)

If I add a specific tag like (only for example: "exampletag") it will not show up in search results if searching for the tag:
![image](https://github.com/hoarder-app/hoarder/assets/80603288/bf5c3f02-37ff-4c19-a872-1d396817b8ad)

Instead I need to move to tags section for having a look for "exampletag":
![image](https://github.com/hoarder-app/hoarder/assets/80603288/2fcfc8df-0e7d-468c-9c1d-76a200a4bd99)

In other words, I'd love to see this result as well at full text search.
This way I could just add some more category tags for each bookmark what raises the chance to find the page what I'm looking for.  I hope you get my point

**Reply by MohamedBassem** [reply]
@xFFF6 thanks the detailed explaination. Technically speaking, we index the tags attached to the bookmarks and you should have seen the bookmarks with "exampletag" on them appear in the full text search result. However, there might be some places were we forgot to re-index the bookmarks when the tags change. This sounds like a bug.

For now, you can fix it by doing a re-index in the admin panel. We will need to check the places were we attach tags and ensure that re-indexing happens everywhere there.

**Comment by xFFF6** [comment]
A re-index did the trick:
![image](https://github.com/hoarder-app/hoarder/assets/80603288/deac7570-18d6-464e-9000-f5d43f771a59)

Are you following up on this, or should I hit "create issue from discussion"?

**Reply by kamtschatka** [reply]
Create an issue please, so i can track it
